depthai_ros_driver/include/depthai_ros_driver/conversion.hpp

```diff
@@ -84,6 +84,30 @@ cv::Mat planar2interleaved(const cv::Mat& mat);
 
 cv::Mat interleaved2planar(const cv::Mat& mat);
 
+/**
+ * @brief Compute the difference between ros::Time::now() and ros::SteadyTime::now()
+ * 
+ * @return ros::Duration 
+ */
+ros::Duration delta_ros_time_to_steady_time(){
+    auto ros_time = ros::Time::now();
+    auto steady_time = ros::SteadyTime::now();
+    auto steady_time_as_ros_time = ros::Time(steady_time.sec, steady_time.nsec);
+
+    return ros_time - steady_time_as_ros_time;
+}
+/**
+ * @brief Convert dai timestamp to ros::Time
+ * 
+ * @param ts dai timestamp 
+ * @return ros::Time 
+ */
+ros::Time timestamp_dai2ros(const depthai_common_msgs::Timestamp& ts) {
+    static ros::Duration delta = delta_ros_time_to_steady_time();
+    // dai uses steady_clock, so add delta to convert it to ros time
```

Comment:
Just a question. Does DepthAI read steady clock of the host PC, and make it a device time?

Reply:
The camera doesn't have a clock. So, when they receive image, they assigned a timestamp in this line of depthai-core -> https://github.com/luxonis/depthai-core/blob/d53ab26/src/device/DeviceBase.cpp#L561